src/main/java/javaserver/ResponseBuilder.java

```diff
@@ -24,19 +24,34 @@ private void setResponseData(String statusLine, String header, String body) {
 
 	private String getStatusLine() {
 		HTTPStatusCode responseCode;
-		if (isSupported(request)) {
-			responseCode = HTTPStatusCode.TWO_HUNDRED;
+		if (requestIsSupported()) {
+			if (RequestParser.isRedirect(request)) {
+				responseCode = HTTPStatusCode.THREE_OH_TWO;
+			} else {
+				responseCode = HTTPStatusCode.TWO_HUNDRED;
+			}
+		} else if (isNotAllowed(request)) {
+			responseCode = HTTPStatusCode.FOUR_OH_FIVE;
 		} else {
 			responseCode = HTTPStatusCode.FOUR_OH_FOUR;
 		}
 		return responseCode.getStatusLine();
 	}
-	
-	private boolean isSupported(Request request) {
+
+	private boolean requestIsSupported() {
 		String requestType = request.getMethod();
-		if (request.routeOptions() != null && (Arrays.asList(request.routeOptions()).contains(requestType))) {
+		if (getRouteOptions() != null && (Arrays.asList(getRouteOptions()).contains(requestType))) {
```

Comment:
Not completely clear at first glance what this conditional means. I think a method should be extracted out here.